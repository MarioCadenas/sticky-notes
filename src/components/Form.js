import React, { Component } from 'react';
import styled from 'styled-components';

const StyledForm = styled.form`
  height: 100px;
  width: 100%;
`;
const Input = styled.input`
  border: none;
  border-bottom: 1px solid grey;
  height: 30px;
  padding: 10px;
  margin: 10px;
  width: 200px;
`;
const Button = styled.button`
  border: none;
  height: 50px;
  padding: 10px;
  margin: 10px;
  width: 200px;
`;

class Form extends Component {
  constructor(props) {
    super(props);
    this.state = {
      text: '',
      showError: false
    }
  }

  handleInputChange = ({ target: { value: text } }) => this.setState({ text, showError: false });

  handleSubmit = (e) => {
    e.preventDefault();
    const { state: { text } } = this;

    if (!text) {
      return this.setState({ showError: true });
    }

    this.setState({ text: '' });
    return this.props.addNote({ text });
  }

  render() {
    const { handleSubmit, handleInputChange, state: { text, showError } } = this;

    return (
      <StyledForm onSubmit={handleSubmit}>
        {showError && <p>You must provide a text</p>}
        <Input type="text" name="note" onChange={handleInputChange} value={text} placeholder="Text for the note" />
        <Button>Add</Button>
      </StyledForm>
    )
  }
}

export default Form;
