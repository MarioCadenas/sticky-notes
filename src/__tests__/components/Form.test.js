import React from 'react';
import { shallow } from 'enzyme';
import Form, { StyledForm } from '../../components/Form';

describe('Form component', () => {
  it('should correctly render Form component', () => {
    const wrapper = shallow(<Form addNote={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
  it('should correctly call addNote if text is not empty', () => {
    const addNote = jest.fn();
    const text = { text: 'test' };
    const wrapper = shallow(<Form addNote={addNote}/>);
    wrapper.setState(text);
    expect(wrapper.find(StyledForm).length).toEqual(1);
    wrapper.find(StyledForm).simulate('submit', { preventDefault: () => {}});
    expect(addNote).toHaveBeenCalledWith(text);
  });
  it('should Not call addNote if text is empty', () => {
    const addNote = jest.fn();
    const wrapper = shallow(<Form addNote={addNote}/>);
    wrapper.setState({ text: '' });
    expect(wrapper.find(StyledForm).length).toEqual(1);
    wrapper.find(StyledForm).simulate('submit', { preventDefault: () => {}});
    expect(addNote).not.toHaveBeenCalled();
    expect(wrapper.find('p').length).toBe(1);
    expect(wrapper.find('p').text()).toBe('You must provide a text');
  });
});