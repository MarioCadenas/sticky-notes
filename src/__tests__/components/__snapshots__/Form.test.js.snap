// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component should correctly render Form component 1`] = `
<styled.form
  onSubmit={[Function]}
>
  <styled.input
    name="note"
    onChange={[Function]}
    placeholder="Text for the note"
    type="text"
    value=""
  />
  <styled.button>
    Add
  </styled.button>
</styled.form>
`;
